src/mudlet.cpp

```diff
@@ -4483,26 +4400,107 @@ void mudlet::setNetworkRequestDefaults(const QUrl& url, QNetworkRequest& request
+        hostNameToTabMap.insert(mpTabBar->tabData(i).toString(), i);
+    }
+    QString oldActiveHostName;
+    if (mpCurrentActiveHost) {
+        oldActiveHostName = mpCurrentActiveHost->getName();
+    }
+
+    if (!pHost || !pHost->mpConsole) {
+        // Ah, we do not seem to have a profile anymore:
+        mpCurrentActiveHost = nullptr;
+        // Nothing else to do if the host to activate doesn't exist
+        return;
+    }
+
+    if (mpCurrentActiveHost && mpCurrentActiveHost == pHost) {
+        // Nothing to do if the caller was the current foreground host, but on
+        // the initial start up mpCurrentActiveHost WILL be a nullptr
         return;
     }
 
     const QString newActiveHostName{pHost->getName()};
-    if (mpCurrentActiveHost != pHost) {
-        // Need to switch mpCurrentActiveHost to be the given pHost and change
-        // the tab to match - without triggering slot_tabChanged(int)
-        int tabToBeActive = mpTabBar->tabIndex(newActiveHostName);
-        if (tabToBeActive >= 0) {
-            mpTabBar->blockSignals(true);
-            mpTabBar->setCurrentIndex(tabToBeActive);
-            mpTabBar->blockSignals(false);
+    int newActiveTabIndex = hostNameToTabMap.value(newActiveHostName, -1);
+
+    if (mpCurrentActiveHost && mpCurrentActiveHost->mpConsole) {
+        // Tell the old profile that it is losing focus:
+        TEvent focusLostEvent {};
+        focusLostEvent.mArgumentList << QLatin1String("sysProfileFocusChangeEvent");
+        // Boolean arguments are carried as "0" for false or "1" for true,
+        // This is for the profile that is losing focus:
+        focusLostEvent.mArgumentList << QLatin1String("0");
+        focusLostEvent.mArgumentTypeList << ARGUMENT_TYPE_STRING << ARGUMENT_TYPE_BOOLEAN;
+        mpCurrentActiveHost->raiseEvent(focusLostEvent);
+
+        if (!mMultiView) {
+            // We only have to hide the current profile under the tab if NOT
+            // in multi-view mode:
+            mpCurrentActiveHost->mpConsole->hide();
         }
-        mpCurrentActiveHost = pHost;
-        updateDiscordNamedIcon();
-        dactionInputLine->setChecked(mpCurrentActiveHost->getCompactInputLine());
-        pHost->updateDisplayDimensions();
-        emit signal_profileActivated(pHost, tabToBeActive);
     }
+
+    // Need to change to the right tab without triggering slot_tabChanged(int)
+    // as that might even be the caller of this method!
+    if (newActiveTabIndex >= 0) {
+        mpTabBar->blockSignals(true);
+        mpTabBar->setCurrentIndex(newActiveTabIndex);
+        mpTabBar->blockSignals(false);
+    }
+
+    mpCurrentActiveHost = pHost;
+    mpCurrentActiveHost->mpConsole->show();
+    mpCurrentActiveHost->mpConsole->repaint();
+    mpCurrentActiveHost->mpConsole->refresh();
+    mpCurrentActiveHost->mpConsole->mpCommandLine->repaint();
+    // CHECKME: Although we no longer force the focus to the profiles main
```

Comment:
Who can check this?

Reply:
Well, I vaguely recall observing that the main command line was expanding/contracting (vertically) in a manner that did not always match up with what it contained - so presumably that would be affected by such repaint operations. If that is indeed the case then it could also apply to secondary / alternative command lines, so I'd hope that anyone who might be investigating that would note this...

Reply:
A comment on the relevant issue would be much easier to find in that case!

Reply:
I haven't found an issue for it (yet) though - and I am not sure if what I saw constituted an actual issue. If there IS one, then steps to address it might need to include all command lines.

Reply:
Alright - if it was an actual problem, mentioning this in the real issue would make it easy to find when someone was going to tackle it.

But like this, where we're not even certain if we have a real problem, I don't see how it'll help anyone much at all.

Reply:
Okay - I've shortened it to a one-liner comment that I think conveys the fundamentals... :stuck_out_tongue: